src/Orleans.Serialization/Invocation/Response.cs

```diff
@@ -170,6 +177,101 @@ public override void Dispose()
         public override string ToString() => _result is { } r ? r.ToString() : "[null]";
     }
 
+    /// <summary>
+    /// Supports raw serialization of <see cref="Response{TResult}"/> values.
+    /// </summary>
+    public abstract class ResponseCodec
+    {
+        public abstract void WriteRaw<TBufferWriter>(ref Writer<TBufferWriter> writer, object value) where TBufferWriter : IBufferWriter<byte>;
+        public abstract object ReadRaw<TInput>(ref Reader<TInput> reader, scoped ref Field field);
+    }
+
+    [RegisterSerializer]
+    internal sealed class PooledResponseCodec<TResult> : ResponseCodec, IFieldCodec<Response<TResult>>
+    {
+        private readonly Type _codecFieldType = typeof(Response<TResult>);
+        private readonly Type _resultType = typeof(TResult);
```

Comment:
Is there a reason to not make these static?

Reply:
Accessing static fields on generic types with reference type parameters is relatively slow (through a dictionary).